app/controllers/congress_messages_controller.rb

```diff
@@ -0,0 +1,101 @@
+class CongressMessagesController < ApplicationController
+  include Tooling
+
+  before_action :set_congress_message_campaign
+  before_action :update_user, only: :create
+  before_action :subscribe_user, only: :create
+
+  rescue_from SmartyStreets::AddressNotFound, with: :address_not_found
+  rescue_from CongressForms::RequestFailed, with: :congress_forms_request_failed
+
+  def new
+    if @campaign.target_bioguide_ids.present?
+      bioguide_ids = @campaign.target_bioguide_ids.split
+    else
+      location = SmartyStreets.get_location(params["street_address"], params["zipcode"])
+      members = @campaign.targets.for_district(location.state, location.district)
+      bioguide_ids = members.pluck(:bioguide_id)
+    end
+
+    forms = CongressForms::Form.find(bioguide_ids)
+    @message = CongressMessage.new_from_lookup(location, params[:message], @campaign, forms)
+    render partial: "form"
+  end
+
+  def create
+    @message = CongressMessage.new(congress_message_params.merge(campaign: @campaign))
+    @message.forms = CongressForms::Form.find(params["bioguide_ids"].split)
+
+    if @message.submit
+      @name = user_params[:first_name] # for deliver_thanks_message
+      track_action
+      deliver_thanks_message
+      render partial: "tools/share"
+    else
+      render plain: I18n.t(:submission_failed, scope: :congress_forms), status: :bad_request
```

Comment:
In the current version, `500`s from the Congress Forms API submit endpoint fail silently. This shows an error, but we might want to preserve the old behavior since the attempted fills will still hopefully be saved in the delayed job queue there. Thoughts?

Reply:
Yes, failed messages are retried periodically, in case an update is made in [unitedstates/contact-congress](https://github.com/unitedstates/contact-congress) which will let them succeed. Submissions can also be rejected for reasons that aren't fixable, but we can't differentiate the two failure types currently.

Since these are usually form messages written by EFF I think it's not the worst thing to swallow errors and make a best effort at delivering the statement later. It's more problematic if the message is fully authored by the user, as it was on democracy.io, or sometimes might be on act.eff.org. In that case it might be worth showing some kind of notification and providing a link directly to the senate.gov contact page?

Reply:
Maybe we could send an e-mail notifying the user and linking them to the senate.gov page. That way submission could still happen asynchronously.

Reply:
:+1: I like that solution.